Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -162,12 +190,40 @@ class PurchasesOrchestrator {
         }
     }
 
-    func purchase(product: SKProduct,
+    func purchase(package: Package, completion: @escaping PurchaseCompletedBlock) {
+        // todo: clean up, move to new class along with the private funcs below
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *),
+           package.productDetails is SK2ProductDetails {
+            _ = Task<Void, Never> {
+                let result = await purchase(sk2Package: package)
+                DispatchQueue.main.async {
+                    switch result {
+                    case .failure(let error):
+                        completion(nil, nil, error, false)
+                    case .success(let (customerInfo, userCancelled)):
+                        // todo: change API and send transaction
+                        if userCancelled {
+                            completion(nil, nil, ErrorUtils.purchaseCancelledError(), userCancelled)
+                        } else {
+                            completion(nil, customerInfo, nil, userCancelled)
+                        }
+                    }
+                }
+            }
+        } else {
+            guard package.productDetails is SK1ProductDetails else {
+                fatalError("could not identify StoreKit version to use!")
```

Comment:
Maybe add `package.productDetails` to the `fatalError` so we can see what it actually was?